src/mcp/client/stdio/__init__.py

```diff
@@ -179,18 +179,18 @@ async def stdin_writer():
             yield read_stream, write_stream
         finally:
             # Clean up process to prevent any dangling orphaned processes
-            try:
-                if sys.platform == "win32":
-                    await terminate_windows_process(process)
-                else:
+            if sys.platform == "win32":
+                await terminate_windows_process(process)
+            else:
+                # On Linux, terminating a process that has already exited raises ProcessLookupError.
+                # This can happen if the command failed to launch properly (as in this test case).
+                try:
                     process.terminate()
-            except ProcessLookupError:
-                # Process already exited, which is fine
-                pass
+                except ProcessLookupError:
+                    # Process already exited — safe to ignore
+                    pass
             await read_stream.aclose()
             await write_stream.aclose()
-            await read_stream_writer.aclose()
-            await write_stream_reader.aclose()
```

Comment:
I don't think we should be removing these 2 `write.aclose()` lines or modifying this file at all - was this necessary to make tests pass or something? Or a potential leftover from the rebase?